api/provider/aws/service_read.go

```diff
@@ -1,45 +1,68 @@
 package aws
 
 import (
-	"fmt"
-
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/errors"
 	"github.com/quintilesims/layer0/common/models"
 )
 
 func (s *ServiceProvider) Read(serviceID string) (*models.Service, error) {
-	model := &models.Service{}
-
-	clusterName := addLayer0Prefix(s.Config.Instance(), model.EnvironmentID)
-	service, err := s.readService(clusterName, serviceID)
+	environmentID, err := lookupEntityEnvironmentID(s.TagStore, "service", serviceID)
 	if err != nil {
 		return nil, err
 	}
 
-	model.DesiredCount = aws.Int64Value(service.DesiredCount)
-	model.RunningCount = aws.Int64Value(service.RunningCount)
-	model.PendingCount = aws.Int64Value(service.PendingCount)
+	fqEnvironmentID := addLayer0Prefix(s.Config.Instance(), environmentID)
+	clusterName := fqEnvironmentID
+	ecsService, err := s.readService(clusterName, serviceID)
+	if err != nil {
+		return nil, err
+	}
 
-	for _, deploy := range service.Deployments {
+	var deployments []models.Deployment
+	for _, deploy := range ecsService.Deployments {
 		deployID := aws.StringValue(deploy.TaskDefinition)
+
 		//todo: convert deployid to layer0 deploy id
 
-		deploy := models.Deployment{
-			DeploymentID: aws.StringValue(deploy.Id),
-			Created:      aws.TimeValue(deploy.CreatedAt),
-			Updated:      aws.TimeValue(deploy.UpdatedAt),
-			Status:       aws.StringValue(deploy.Status),
-			PendingCount: aws.Int64Value(deploy.PendingCount),
-			RunningCount: aws.Int64Value(deploy.RunningCount),
-			DesiredCount: aws.Int64Value(deploy.DesiredCount),
-			DeployID:     deployID,
+		d := models.Deployment{
+			Created:       aws.TimeValue(deploy.CreatedAt),
+			DeployID:      deployID,
+			DeployName:    "", // tag
+			DeployVersion: "", // tag
+			DesiredCount:  int(aws.Int64Value(deploy.DesiredCount)),
+			PendingCount:  int(aws.Int64Value(deploy.PendingCount)),
+			RunningCount:  int(aws.Int64Value(deploy.RunningCount)),
+			Status:        aws.StringValue(deploy.Status),
+			Updated:       aws.TimeValue(deploy.UpdatedAt),
 		}
 
-		model.Deployments = append(model.Deployments, deploy)
+		deployments = append(deployments, d)
+	}
+
+	var loadBalancerID string
+	if len(ecsService.LoadBalancers) != 0 {
+		loadBalancer := ecsService.LoadBalancers[0]
+		loadBalancerName := aws.StringValue(loadBalancer.LoadBalancerName)
+		fqLoadBalancerID := loadBalancerName
+		loadBalancerID = delLayer0Prefix(s.Config.Instance(), fqLoadBalancerID)
+	}
+
+	model := &models.Service{
+		Deployments:      deployments,
+		DesiredCount:     int(aws.Int64Value(ecsService.DesiredCount)),
+		EnvironmentID:    environmentID,
+		EnvironmentName:  "", // tag
```

Comment:
I think calling out which fields get tags is a step in the right direction. I'm hoping we can think of a way to make the relationship even clearer, but that is out of scope for this pr.